File: array/867_transpose.java
class Solution {
    public int[][] transpose(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        int[][] ans = new int[n][m];  // as rows and columns are interchanged so n,m size of ans array
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                ans[j][i] = matrix[i][j]; // swap the indices yk
            }
        }
        return ans;
    }
}

// Time-complexity: O(N^2)
// Space-complexity: O(N^2)

File: array/867__transpose.py
class Solution(object):
    def transpose(self, matrix):     
        m=len(matrix)
        n=len(matrix[0])
        A=[[0]*m for _ in range(n)]  #to assign nested listed to A and assign each row element as 0
        for i in range (0,m):
            for j in range(0,n):
                A[j][i]=matrix[i][j] #cause in logic we are using assignment operator, to use it need to have initial value
        return A
    
# Time-complexity: O(N^2)
# Space-complexity: O(N^2)